src/controller/rpc.cpp

```diff
@@ -0,0 +1,208 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2020 - 2022 Pionix GmbH and Contributors to EVerest
+
+#include "rpc.hpp"
+
+#include "command_api.hpp"
+#include "ipc.hpp"
+
+enum class RPCRequestErrorCode
+{
+    ParseError = -32700,
+    InvalidRequest = -32600,
+    MethodNotFound = -32601,
+    InvalidParams = -32602,
+    InternalError = -32603
+};
+
+class RPCRequestError : public std::runtime_error {
```

Comment:
What's the reason for defining this class directly in the rpc.cpp not in the rcp.hpp ?

Reply:
Probably because no one else is using it, I guess